lib/health_cards/health_card.rb

```diff
@@ -1,64 +1,208 @@
+      # @param ver [String] FHIR Version supported by this HealthCard (sub)class. Leaving this param out
+      # will only return the current value
+      # value (used as a getter)
+      # @return [String] Current FHIR version supported
+      def fhir_version(ver = nil)
+        @fhir_version ||= ver unless ver.nil?
+        @fhir_version
+      end
+
+      # Additional type claims this HealthCard class supports
+      # @param types [String, Array] A string or array of string representing the additional type claims
+      def additional_types(*types)
+        @types ||= VC_TYPE
+        @types += types
+      end
+
+      # Type claims supported by this HealthCard subclass
+      # @return [Array] an array of Strings with all the supported type claims
+      def types
+        @types ||= VC_TYPE
+      end
+    end
 
     # Create a HealthCard
     #
     # @param verifiable_credential [HealthCards::VerifiableCredential] VerifiableCredential containing a fhir bundle
     # @param jws [HealthCards::JWS] JWS which should have a payload generated from the verifiable_credential
-    def initialize(verifiable_credential: nil, jws: nil)
-      self.verifiable_credential = verifiable_credential
-      @jws = jws
+    def initialize(bundle:, issuer: nil)
+      raise InvalidPayloadException unless bundle.is_a?(FHIR::Bundle)
+
+      @issuer = issuer
+      @bundle = bundle
     end
 
-    # Set the HealthCard payload
-    #
-    # @param new_payload [HealthCards::VerifiableCredential, String] the FHIR bundle used as the Health Card payload
-    def verifiable_credential=(new_payload)
-      raise InvalidPayloadException unless new_payload.nil? || new_payload.is_a?(HealthCards::VerifiableCredential)
+    # A Hash matching the VC structure specified by https://smarthealth.cards/#health-cards-are-encoded-as-compact-serialization-json-web-signatures-jws
+    # @return [Hash]
+    def to_hash
+      {
+        iss: issuer,
+        nbf: Time.now.to_i,
+        vc: {
+          type: self.class.types,
+          credentialSubject: {
+            fhirVersion: self.class.fhir_version,
+            fhirBundle: strip_fhir_bundle
+          }
+        }
 
-      @jws.payload = new_payload&.compress_credential if @jws
-      @verifiable_credential = new_payload
+      }
     end
 
-    # Save the HealthCard as a file
-    #
-    # @param file_name [String] the name of the file
-    def save_to_file(file_name)
-      File.open(file_name, 'w') do |file|
-        file.write(to_json)
-      end
+    # A compressed version of the FHIR::Bundle based on the SMART Health Cards frame work and any other constraints
+    # defined by a subclass
+    # @return String compressed payload
+    def to_s
+      HealthCard.compress_payload(to_json)
+    end
+
+    # Chunks the compress payload for us in generating QR codes
+    # @return [Array] An array of strings used to create QR codes
+    def chunks
+      HealthCards::Chunking.generate_qr_chunks to_s
```

Comment:
Should we just `include` this module?

Reply:
Maybe? I'm not even sure why it's a module as it's only used in health card

Reply:
If we do want to change how we do this though, I'd consider that out of scope for this task.